### ISSUE TITLE
Flutter drive --browser-dimension not working

### ISSUE THREAD

**Issue by hoantruong** [flutter/flutter#136109]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

I am trying to run flutter web test on Chrome, with the following command:

```
flutter drive \
  --driver=test_driver/integration_test.dart \
  --target=integration_test/main_test.dart > C:\Automation_test\Report_log\log.txt \
  --browser-dimension="1920,1080" \
  -d chrome
```

### Expected results

The browser is resized to 1920 x 1080 when start.

### Actual results

The browser is NOT resized to 1920 x 1080 when start.

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I verified using a sample app and running using latest versions and observed that passing browser dimension is not taking effect. I tried passing different dimensions, but the browser window size is same:

1. 1920 x 1080:

<img width="1423" alt="Screenshot 2023-10-09 at 1 42 17 PM" src="https://github.com/flutter/flutter/assets/67046386/621273b7-8e73-4102-871b-4880f07e42fe">


2. 400 x 800:

<img width="1440" alt="Screenshot 2023-10-09 at 1 43 44 PM" src="https://github.com/flutter/flutter/assets/67046386/567c6785-4d1c-4c09-8853-7dcb29883499">


3. Without passing `--browser-dimension` argument:

<img width="1433" alt="Screenshot 2023-10-09 at 1 45 14 PM" src="https://github.com/flutter/flutter/assets/67046386/fc524d53-b508-4791-8a4c-da18aae62bca">

**Comment by hoantruong** [comment]
Could you please help to fix this issue soon @darshankawar? 

Thanks very much!

**Comment by amrgetment** [comment]
Yes, I can confirm it still exists on the latest Flutter release 3.24.5
is there any workaround

**Comment by amrgetment** [comment]
@kevmoo Sorry to mention you, but could you look into this issue, it is breaking the visual UI testing(golden testing) for Flutter web

**Comment by munrocket** [comment]
We need to add mobile emulation in dart.

Here same bug with selenium and fix for it.
```js
const { Builder } = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');
const fs = require('fs');

const options = new chrome.Options();
options.addArguments('--no-sandbox');

// FIX BUG ==>
options.setMobileEmulation({
    deviceMetrics: {
        width: 393,
        height: 852,
        pixelRatio: 3.0 // iPhone 15 size
    },
    userAgent: 'Mozilla/5.0 (Linux; Android 11; Pixel 6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.91 Mobile Safari/537.36'
});

const driver = new Builder()
    .forBrowser('chrome')
    .usingServer('http://localhost:4444')
    .setChromeOptions(options)
    .build();

(async function resize() {
    try {
        await driver.manage().window().setRect({ width: 393, height: 852 });
        if (!fs.existsSync('screenshots')) {
            fs.mkdirSync('screenshots');
        }
        const screenshot = await driver.takeScreenshot();
        fs.writeFileSync('screenshots/screenshot.png', screenshot, 'base64');
    } finally {
        await driver.quit();
    }
})();
```

**Comment by munrocket** [comment]
@amrgetment I am created [pull request](https://github.com/flutter/flutter/pull/160678). You can apply [patch](https://pastebin.com/JtV1yfdb) to stable Flutter with `git apply patch.diff`, don't forget to clean ~/.pub-cache before use, if it is not clean setup.

**Comment by ksokolovskyi** [comment]
Hi @darshankawar, not long ago this issue was closed as fixed by https://github.com/flutter/flutter/pull/160678. While investigating another driver issue on `master` channel, I noticed that passing `--browser-dimensions` doesn't resize the browser window.

I am using `flutter-dev` to run Flutter Tools from source.

#### With `--browser-dimension='200x300@1'`

```
flutter-dev drive \
   --driver=test_driver/integration_test.dart \
   --target=integration_test/app_test.dart \
   -d chrome --browser-dimension='200x300@1'
```

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/cabcf104-8720-45b5-8bcf-39f10eb7e4d5" />

#### Without `--browser-dimension`

```
flutter-dev drive \
   --driver=test_driver/integration_test.dart \
   --target=integration_test/app_test.dart \
   -d chrome
```

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/d4586707-6bad-4262-a2c9-f7e4ef891820" />

⠀

Could you please try to reproduce the issue again on `master` channel? Thanks in advance!
